### PR TITLE
Imbalance fees

### LINKED ISSUE #2577: I want to take imbalance penalty mediation fees SDK and CLI
## User Story
As Kelsos I want to take imbalance penalty mediation fees so that I am able to incentive an optimal channel balance through fees while mediating transfers.

## Acceptance criteria
- Imbalance penalty mediation fees support on SDK on while meditating
- CLI exposes config options through parameters
- MFEE3-4 scenarios passing

## Tasks
- [x] Define and test a pure function of in & out channels states and config, which returns the final fee after imbalance penalties
- [x] Expose and document options on config interface and CLI

### PR DESCRIPTION
**Pull request by andrevmatos**
Fixes #2577 

**Short description**
Adds imbalance fees to SDK and CLI.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. MFEE3 & MFEE4 pass

### CODE REVIEW

File: raiden-ts/src/transfers/mediate/types.ts [hunk 1]
```diff
@@ -54,44 +65,290 @@ export interface FeeModel<Config, Schedule> {
...
+
+  if (proportionalImbalanceFee === 0) {
+    return [
+      [Zero as UInt<32>, Zero as UInt<32>],
+      [channelCapacity as UInt<32>, Zero as UInt<32>],
+    ];
+  }
+  const bnProportionalImbalanceFee = new Decimal(proportionalImbalanceFee).div(1e6);
+
+  const MAX_SLOPE = new Decimal('0.1');
```
Comment (palango) [inline raiden-ts/src/transfers/mediate/types.ts:299]: nit: Please move that to the beginning of the function

**Review by palango — COMMENTED** [review]
Really nice PR. But working with big numbers (and now big decimals) is a pain in the ass. To bad there's no operator overloading.

I'll run the scenarios now, but there some integration tests that fail.

We also must not forget to set the defaults, but that should be done in a different PR.

File: raiden-ts/tests/setup.ts [hunk 2]
```diff
@@ -17,3 +18,11 @@ Object.defineProperty(BigNumber.prototype, util.inspect.custom, {
     return `${opts.stylize('BN', 'special')}(${opts.stylize(this.toString(), 'number')})`;
```
[the remark this answers is not in the document]
Comment (andrevmatos) [reply raiden-ts/tests/setup.ts:18]: No. ether's `BigNumber` is still used all over the place. It's actually still the BigNumber library we use in the codebase. `decimal.js` was added only for this specific mediation calculation, but is converted to the former before reaching "public" interface. I added the styling option here to make mediation test errors more readable.

File: raiden-ts/src/transfers/mediate/types.ts [hunk 3]
```diff
@@ -54,44 +65,290 @@ export interface FeeModel<Config, Schedule> {
...
+    // flat fee just sums once for each channel
+    return () => decode(Int(32), flat.mul(2));
   },
   schedule(flat) {
     return { flat };
   },
 };
 
-export const proportionalFee: FeeModel<Int<32>, { proportional: Int<32> }> = {
+export const proportionalFee: FeeModel<number, { proportional: Int<32> }> = {
```
[the remark this answers is not in the document]
Comment (andrevmatos) [reply raiden-ts/src/transfers/mediate/types.ts:159]: Proportional fees should be < 1e6 (100%), so there's no need for its config to be a BigNumber. It was a leftover from before, and I fixed it here to make it simpler with a simple number. This is internal anyway, as this type is just the type which gets handled as `unknown` but passed to the rest of the methods after decoding/validating here.

File: raiden-ts/src/transfers/mediate/types.ts (hunk 1 above)
Comment (andrevmatos) [reply raiden-ts/src/transfers/mediate/types.ts:299]: Sure. It went here because I replicated PC's respective function, but I'll move. Thanks

**Comment by andrevmatos** [comment]
MFEE4 passing on Rinkeby on `next` based on top of this PR, with workaround from https://github.com/raiden-network/raiden/issues/7130
Screenshot_20210607_214820

**Comment by palango** [comment]
With https://github.com/raiden-network/raiden/pull/7132 I have green local runs on mfee3 and mfee4

**Review by palango — APPROVED** [review]
Great! Green scenarios finally! 🎉

**Pull request merged**